### ISSUE TITLE
[Bug] Prompt content error when subscription fails

### ISSUE THREAD

**Issue by Rory-Z** [emqx/MQTTX#527]
#### What did I do

I use emqx broker and refuse to subscribe to all topics. When I use mqttx to subscribe, it prompts me to refuse to subscribe to the $SYS topic

#### Expected

Correct prompt content

#### Environment

- OS: MacOS 10.15.7
- MQTTX version: v1.4.2

#### More detail
![15761614670651_ pic](https://user-images.githubusercontent.com/16801068/109619381-b00c8c80-7b73-11eb-9a98-ac6bc9f02bbb.jpg)

**Comment by oceanlvr** [comment]
@Rory-Z Hi! Just like we talk before, I think we should keep this prompt.
就像我们之前讨论那样，我认为应该保留这个prompt

**TL;DR: This prompt won't effect `MQTT X` to subscript other `MQTT broker`(which is not `EMQX`).**
prompt 提示不会影响 MQTT X 订阅其他 MQTT broker的行为（不是 EMQX的）


The fellow is my thought:

I draw a picture for our `MQTT X` subscription:
![image](https://user-images.githubusercontent.com/36698124/109672512-144d4180-7bb0-11eb-8075-55b1e246ad22.png)

![image](https://user-images.githubusercontent.com/36698124/109676661-e4a03880-7bb3-11eb-9652-f6218f1eeb7c.png)


When you get the prompt `refuse to subscribe to the $SYS topic`, it only could be rejected by `ACL`, not the `MQTT` protocol normal reject. (`subscript begin` -> `subscript MQTT broker success?` -> `reject by MQTT ACL` -> `return Qos 128
(not 1|2|3)`)

当你收到'拒绝订阅$SYS主题'的时候，它只可能是被ACL拒绝的，而不是标准的 MQTT 拒绝。即流程是(`subscript begin` -> `subscript MQTT broker success?` -> `reject by MQTT ACL` -> `return Qos 128
(not 1|2|3)`)


So if we get this prompt it only could be caused by an abnormal `MQTT` subscription, we don't need to wrong about other broker's Inconsistent behavior.

所以我们得到这个prompt的时候只可能是被非标准的 MQTT 订阅拒绝的，我们不需要担心其他broker的不统一行为。

```js
if (res.length < 1) {
  errorReason = SubscribeErrorReason.emptySubFailed
} else if (![0, 1, 2].includes(res[0].qos) && !topic.match('/^($SYS)//i')) {
// get in here, so it must subscript $SYS topic. but not cause by normal MQTT protocol, so no reject code return.
// and the Qos code is 128
  errorReason = SubscribeErrorReason.qosSubSysFailed
} else if (![0, 1, 2].includes(res[0].qos)) {
// same as above, not cause by normal MQTT protocol.
// and it's a abnormal reject, return qos 128
  errorReason = SubscribeErrorReason.qosSubFailed
}
```
ref:
https://www.hivemq.com/blog/mqtt-essentials-part-4-mqtt-publish-subscribe-unsubscribe/
https://github.com/mqttjs/MQTT.js/issues/728



cc @ysfscream  I think we should not change any thing unless `MQTT.js` update, what's your opinion?

**Comment by oceanlvr** [comment]
https://github.com/mqttjs/MQTT.js/blob/37b12cb94737acddfe0fa016e013516ba3b9a031/bin/sub.js#L89-L102

**Comment by oceanlvr** [comment]
![image](https://user-images.githubusercontent.com/36698124/109738360-a9792600-7c02-11eb-83de-4f93e2ddbf0a.png)
So in a word, ACL reject happened in step 2 `SUBBACK`, it only return `Qos` 128, and normal reject happened in step 1 (by MQTT broker) `SUBSCRIBE`

**Comment by ysfscream** [comment]
@oceanlvr Yes! But I think the problem with this problem is that if sub `test` fails, it cannot be shown that sub `$SYS` failed. Although it should indeed be prompted to check the ACL configuration.

**Comment by oceanlvr** [comment]
@ysfscream  I understand now, Yes, we should not prompt this when we sub `test` failed. Sorry i didn't understand what you mean.
